src/Butil/Bit.Butil/Scripts/userAgent.ts

```
@@ -0,0 +1,595 @@
+        // Ensure OS does not include the browser name.
+        if (os && os.indexOf(name) != -1 && !RegExp(name + ' OS').test(os)) {
+            os = os.replace(RegExp(' *' + qualify(name) + ' *'), '');
+        }
+
+        // Add layout engine.
+        if (layout && !/\b(?:Avant|Nook)\b/.test(name) && (
+            /Browser|Lunascape|Maxthon/.test(name) ||
+            name != 'Safari' && /^iOS/.test(os) && /\bSafari\b/.test(layout[1]) ||
+            /^(?:Adobe|Arora|Breach|Midori|Opera|Phantom|Rekonq|Rock|Samsung Internet|Sleipnir|SRWare Iron|Vivaldi|Web)/.test(name) && layout[1])) {
+            // Don't add layout details to description if they are falsey.
+            (data = layout[layout.length - 1]) && description.push(data);
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)